FeatureExtractor._fill_vnode_parents() bug

Sometimes you get a parent UAST node for the vnode wrongly.
Example:
```
AssertionError: 
Wrong parent of j=3 
node=VirtualNode(' Polyfill ES6 Promise', y=None, start=(2, 1, 3), end=(23, 1, 24), node=4747332400, path="test/specs/__helpers.js"). 
parent.start_position.offset=24, parent.end_position.offset=58
```
So parent does not cover original UAST Node by offsets.

How to reproduce:
Take this PR: https://github.com/src-d/style-analyzer/pull/664
Run
```python
python3 -m lookout.style.format quality-report -o ./delme -i ./lookout/style/format/benchmarks/data/quality_report_repos_with_vnodes_number.csv
```
in the logs you see
```
ERROR:9088:EventListener:FAIL 3.149
Traceback (most recent call last):
  File "/Users/k/sourced/workdir/lookout-sdk-ml/lookout/core/event_listener.py", line 184, in wrapped_catch_them_all
    return func(self, request, context)
  File "/Users/k/sourced/workdir/lookout-sdk-ml/lookout/core/event_listener.py", line 209, in wrapped_handle
    return getattr(self.handlers, method_name)(request)
  File "/Users/k/sourced/workdir/lookout-sdk-ml/lookout/core/manager.py", line 109, in process_push_event
    model = analyzer.train(ptr, mycfg, data_service)
  File "/Users/k/sourced/workdir/style-analyzer/lookout/style/format/benchmarks/general_report.py", line 200, in train
    super().train(ptr, config, data_service)
  File "/Users/k/sourced/workdir/lookout-sdk-ml/lookout/core/data_requests.py", line 96, in wrapped_handle_rpc_errors
    return func(cls, ptr, config, data_service, **data)
  File "/Users/k/sourced/workdir/lookout-sdk-ml/lookout/core/data_requests.py", line 228, in wrapped_with_uasts_and_contents
    return func(cls, ptr, config, data_service, files=files, **data)
  File "/Users/k/sourced/workdir/style-analyzer/lookout/style/format/analyzer.py", line 209, in train
    X_train, y_train, _ = fe.extract_features(train_files)
  File "/Users/k/sourced/workdir/style-analyzer/lookout/style/format/feature_extractor.py", line 192, in extract_features
    parsed_files, node_parents, vnode_parents = self._parse_vnodes(files, lines)
  File "/Users/k/sourced/workdir/style-analyzer/lookout/style/format/feature_extractor.py", line 330, in _parse_vnodes
    self._fill_vnode_parents(file_parents, file_vnodes, uast, vnode_parents)
  File "/Users/k/sourced/workdir/style-analyzer/lookout/style/format/feature_extractor.py", line 812, in _fill_vnode_parents
    j, repr(vn), parent.start_position.offset, parent.end_position.offset)
AssertionError: Wrong parent of j=3 node=VirtualNode(' Polyfill ES6 Promise', y=None, start=(2, 1, 3), end=(23, 1, 24), node=4747332400, path="test/specs/__helpers.js"). parent.start_position.offset=24, parent.start_position.offset=58
```

--- a/lookout/style/format/feature_extractor.py
+++ b/lookout/style/format/feature_extractor.py
@@ -803,7 +803,8 @@ class FeatureExtractor:
         for j, vn in enumerate(file_vnodes):
             if vn.node:
                 closest_left_node_id = id(vn.node)
-            parent = self._find_parent(j, file_vnodes, file_parents, closest_left_node_id)
-            if parent is None:
-                parent = uast
-            vnode_parents[id(vn)] = parent
+                vnode_parents[id(vn)] = file_parents[id(vn.node)]
+            else:
+                parent = (self._find_parent(j, file_vnodes, file_parents, closest_left_node_id)
+                          or uast)
+                vnode_parents[id(vn)] = parent

--- a/lookout/style/format/uast_stability_checker.py
+++ b/lookout/style/format/uast_stability_checker.py
@@ -101,7 +101,7 @@ class UASTStabilityChecker:
             if vnode.start.offset not in start_offset_to_vnodes:
                 # NOOP always included
                 start_offset_to_vnodes[vnode.start.offset] = i
-        for i, vnode in enumerate(vnodes[::-1]):
+        for i, vnode in enumerate(vnodes[::-1], start=1):
             if vnode.end.offset not in end_offset_to_vnodes:
                 # NOOP always included that is why we have reverse order in this loop
                 end_offset_to_vnodes[vnode.end.offset] = len(vnodes) - i
@@ -141,7 +141,6 @@ class UASTStabilityChecker:
             vnode_start_index = start_offset_to_vnodes[start]
             vnode_end_index = end_offset_to_vnodes[end]
 
-            assert vnode_start_index <= vnodes_i < vnode_end_index
             try:
                 content_after = self._code_generator.generate_one_change(
                     vnodes[vnode_start_index:vnode_end_index],

